.pre-commit-config.yaml

```diff
@@ -0,0 +1,19 @@
+-   repo: https://github.com/pre-commit/pre-commit-hooks
+    sha: master
```

Comment:
Don't use floating commit pointers. Prefer things, which *always* point to the same commit over time. Pinning dependencies is a crucial part of doing software development, it helps to ensure that environment where you develop things matches production and testing/staging/CI systems, otherwise if you come back to the project in a year it may be broken because of pointing to other versions of dependencies and it's a very painful and thankless process of finding out versions of everything in the env which once was "working fine".

And you can also face a common problem called "works on my machine" when not pinning deps:
![](https://cdn.shopify.com/s/files/1/2233/3607/products/mockup-c6161bd3_800x.jpg)

Reply:
`sha: master` - is not a right way, right?
I need some like 
```
repos:
-   repo: https://github.com/pre-commit/pre-commit-hooks
    rev: v1.2.3
```
I right understand?
P.S. 

> Don't use floating commit pointers.

Don't or Do?

Reply:
**do not**

`master` points to the `HEAD` of the branch and once someone pushes new commits it points to the new last commit there. it's redundant.
use commit sha or tag instead

Reply:
From - https://pre-commit.com
![screenshot from 2018-05-31 10-26-36](https://user-images.githubusercontent.com/24910277/40768168-2f6008e8-64bd-11e8-8910-a9b682ff72ab.png)